### PR TITLE
Add ability to specify hostname in Host header for phantomjs/single mode

### PR DESCRIPTION
**Pull request by r0p0s3c**
Hi,

in cases where one is trying to run eyewitness against an IP that serves multiple virtual hosts, being able to set the Host header is useful to request a specific virtual host. While normally you'd expect DNS to be set up correctly, where this is not the case, one needs to run against the IP and specify a Host explicitly. This PR allows that in single+headless mode, as phantomjs makes it straightforward, so submitting this for inclusion

### CODE REVIEW

**Comment by ChrisTruncer** [comment]
Thanks for sending this over to me.  Checking it out now!

File: EyeWitness.py [hunk 1]
```diff
@@ -202,6 +203,10 @@ def create_cli_parser():
         parser.print_help()
         sys.exit()
 
+    if not all((args.vhost_name, args.single, args.headless)):
```
Comment (ChrisTruncer) [inline EyeWitness.py]: There's a bug here with the if statement.  This is causing the program to exit if only providing a file and telling it to use headless (which is a legitimate use) without specifying vhost_name.

Ex: ./EyeWitness.py -f urls.txt --headless
This errors out.  You'll need to just change the if statement to first detect if a vhost is being specified, and if so, make sure it has its requirements (headless and single).

**Comment by ChrisTruncer** [comment]
I just added a comment to the code, let me know if you have the chance to check it out

**Comment by ChrisTruncer** [comment]
Were you still interested in getting this merged in?

**Comment by r0p0s3c** [comment]
Apologies and thanks for the comment, this slipped through the cracks. I've fixed the if statement, should again be ready for a merge.

**Comment by ChrisTruncer** [comment]
Hey,

So I tried running it again now, and it looks like I am getting an error.  I basically ran something along the lines of:

./EyeWitness.py --single http://www.google.com --vhost-name test.com --headless

I've also tried:

./EyeWitness.py --single http://www.google.com --vhost-name test --headless

The error I am getting is:

error

**Comment by r0p0s3c** [comment]
Apologies, looks like a typo. Will fix shortly.

**Comment by r0p0s3c** [comment]
Typo fixed and tested, confirmed working using the arguments used above

**Comment by ChrisTruncer** [comment]
Hmm... I'm not sure why, but this is now having merge conflicts.  This might be due to the recent addition of live credential testing.  Would you mind taking a quick look at this?

Sorry about that.  But this is something I'd definitely like to get added in as I could see a good use case for it.

Thanks again for the help and submitting this feature over.  It will work well and I think help people in similar situations. :)

**Comment by ChrisTruncer** [comment]
Hey, just wanted to check in again on this

**Comment by r0p0s3c** [comment]
Hi Chris,

thanks for the prompt, just merged in the latest changes so PR is conflict free again.

**Comment by ChrisTruncer** [comment]
looks perfect and works in the tests I just ran.  Getting this merged in

**Comment by ChrisTruncer** [comment]
merged, thanks for sending this over!

**Comment by r0p0s3c** [comment]
No worries, thanks for merging

**Pull request merged**